### ISSUE TITLE
EventEmitter memory leak

### ISSUE THREAD

**Issue by ehmicky** [sindresorhus/execa#321]
Steps to reproduce: 

`index.js`:

```js
import execa from 'execa'

(async () => {
  const childProcess = execa('echo', ['test'])
  await childProcess
  await childProcess
  await childProcess
  await childProcess
  await childProcess
})()
```

```bash
$ node index.js
(node:12764) MaxListenersExceededWarning: Possible EventEmitter memory leak detected. 11 close listeners added to [Socket]. Use emitter.setMaxListeners() to increase limit
(node:12764) MaxListenersExceededWarning: Possible EventEmitter memory leak detected. 11 close listeners added to [Socket]. Use emitter.setMaxListeners() to increase limit
```

Node `v12.4.0`
Ubuntu `19.10`
Execa `2.0.0`

This might not be an issue, but simply us setting more than 10 global listeners. But I would like to investigate this to make sure.

**Comment by ehmicky** [comment]
This seems to be a bug with [`pump`](https://github.com/mafintosh/pump), a dependency of [`get-stream`](https://github.com/sindresorhus/get-stream) that we use to retrieve streams with promises. [`Stream.pipeline()`](https://nodejs.org/api/stream.html#stream_stream_pipeline_streams_callback) (available since Node `10.0.0`) is a copy/paste of `pump` so I submitted a bug report [directly to Node](https://github.com/nodejs/node/issues/28432).

**Comment by ehmicky** [comment]
PR at #330.
